### ISSUE TITLE
Problem setting up AMI: kali linux and windows client

### ISSUE THREAD

**Issue by powermundsen** [splunk/attack_range#297]
I get two errors when building attack range with the kali linux and windows client enabled. I'm not to familiar with terraform and the ami setup so this could be a stupid question, but do I have to specify which Ami's to use? 
My region is region = us-west-2.

```
Error: Your query returned no results. Please change your search criteria and try again.

  on modules/kali_machine/resources.tf line 2, in data "aws_ami" "latest-kali-linux":
   2: data "aws_ami" "latest-kali-linux" {



Error: Your query returned no results. Please change your search criteria and try again.

  on modules/windows-client/resources.tf line 2, in data "aws_ami" "windows-client-ami":
   2: data "aws_ami" "windows-client-ami" {
```

**Comment by P4T12ICK** [comment]
The windows client ami needs to be uploaded by yourself because AWS Marketplace doesn't offer Windows 10 amis. We wrote a wiki article about this: https://github.com/splunk/attack_range/wiki/Upload-Windows-10-AMI-to-AWS 

The kali linux ami should work. Let me check, sometimes amis are not avialable anymore in the marketplace and then we need to change them. Then it will still work for the people who used them but are not available anymore for new user.

**Comment by powermundsen** [comment]
Thanks for the quick response!
I missed the wiki on Windows 10, I'll try that. It sounds like you're right concerning the Kali Linux, let me know if you find the correct one :)

**Comment by P4T12ICK** [comment]
I just merged a PR. Can you have a look, if the Kali Linux works for you?

**Comment by powermundsen** [comment]
I have tested the new merge and the search still fails to find the kali linux ami:

`Error: Your query returned no results. Please change your search criteria and try again.
  on modules/kali_machine/resources.tf line 2, in data "aws_ami" "latest-kali-linux":
   2: data "aws_ami" "latest-kali-linux" {
`

I also tried to write the exact name without any luck: 
`  filter {
      name   = "name"
      values = ["Kali Linux 2020.2"]
  }`

**Comment by P4T12ICK** [comment]
The star at the end is important. Can you try to first subscribe to this AMI in Amazon Marketplace and then try it again?

**Comment by powermundsen** [comment]
Thanks for the help! I can't seem to get this to work, I'm still getting the same error. 
I am subscribed to the AMI here https://aws.amazon.com/marketplace/pp/B01M26MMTT?qid=1594358291390&sr=0-1&ref_=srh_res_product_image and the star is present in the search. I also reinstalled the project again but it made no difference. 

`Error: Your query returned no results. Please change your search criteria and try again.
on modules/kali_machine/resources.tf line 2, in data "aws_ami" "latest-kali-linux":
   2: data "aws_ami" "latest-kali-linux" {
`

Do you have a clue as to why this isn't working? Maybe you know the most recent AMI ID, I could try to use directly instead of searching for one? At least that way I'll be able to use attack range as I hoped the next few days.

**Comment by P4T12ICK** [comment]
This is weird. It's hard to help you when I can't reproduce it. It needs to be related that I already subscribed to this AMI a long time ago and you not yet. It's hard to say. 

In the file terraform/modules/kali_machine/resources.tf line number 20 is the ami field. You can try to identify the ami id of the kali machine and manually add it in there.

Maybe some configurations are missing for your aws cli.

Can you try to run the following query:
aws ec2 describe-images --owners 679593333241 --filters "Name=name,Values=Kali Linux 2020*"

you should get two results.

**Comment by powermundsen** [comment]
Yeah I get that it's difficult. Thanks, I'll try to do that.
But I think you're on to something with the was cli. I ran the command with no results. 

`(venv) Andres-MacBook-Pro:attack_range andre$ aws ec2 describe-images --owners 679593333241 --filters "Name=name,Values=Kali Linux 2020*"
{
    "Images": []
}
`
I'm gonna reinstall it now. Are you using was cli version 2.x or version 1.x btw?

**Comment by powermundsen** [comment]
I found the error! 
The name of the machine is wrong, it should be:
`  filter {
      name   = "name"
      values = ["kali-linux-2020.*"]
  }`

**Comment by powermundsen** [comment]
(I opened the issue again to make sure that there are no consequential errors down the line by changing the username and to let you know about the solve, feel free to close it.) 

It can now find an AMI but I run into a SSH error when building: 

`
module.kali_machine.aws_instance.kali_machine[0] (remote-exec): Connecting to remote host via SSH...
module.kali_machine.aws_instance.kali_machine[0] (remote-exec):   Host: 34.220.87.167
module.kali_machine.aws_instance.kali_machine[0] (remote-exec):   User: ec2-user
module.kali_machine.aws_instance.kali_machine[0] (remote-exec):   Password: false
module.kali_machine.aws_instance.kali_machine[0] (remote-exec):   Private key: true
module.kali_machine.aws_instance.kali_machine[0] (remote-exec):   Certificate: false
module.kali_machine.aws_instance.kali_machine[0] (remote-exec):   SSH Agent: true
module.kali_machine.aws_instance.kali_machine[0] (remote-exec):   Checking Host Key: false
module.kali_machine.aws_instance.kali_machine[0]: Still creating... [5m40s elapsed]
Error: timeout - last error: SSH authentication failed (ec2-user@34.220.87.167:22): ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain
`

It seems to be the username "ec2-user" is incorrect and needs to be "kali". When changed to this the build finishes without errors and I can also SSH into the instance through my local terminal. 


In summary, I suggest changing the following in the master: 


attack_range/terraform/modules/kali_machine/resources.tf line 9 ->       `values = ["kali-linux-2020.*"]`
attack_range/terraform/modules/kali_machine/resources.tf line 35 ->       `user        = "kali"`

**Comment by d1vious** [comment]
@powermundsen I would like to keep this issue open a bit longer, maybe it might help anyone if they run into the same issue. Will also talk to new users to see if they run into it. Thank you so much for coming back and posting the full fix! At worst we can easily turn this into a PR from your work. In short 🙇 thank you!

**Comment by mrpappagiorgio** [comment]
I am having a similar issue when building with vagrant and virtualbox locally, not on AWS. ....Pertaining to Kali. I can build everything else but the Kali when i include it. I can Build a Kali box separately via a build though. So it only happens when i build a Kali box as well as a combination of Windows AD and Servers and Clients....It pauses at the SSH as well... increased the Timeout to 900 from 600 and that didn't help...... willing to test out any suggestions.... 

Ubuntu 18.04.4 LTS
Vagrant 2.2.7
Python Dev
Python 3.6.9

**Comment by powermundsen** [comment]
Hi! I forgot to add that I had to change the login user from ”ec2-user” to ”kali” in the configuration file for the kali machine (can't remember what it's called due to vacation right now). That is the reason why it times out! You can also try SSh first in you local terminal to see if that solves the problem. 
This is something that might needs to be corrected for all builds?

**Comment by P4T12ICK** [comment]
This was fixed with the latest merge
